docs/adrs/rework-batch-analysis-architecture.md

```diff
@@ -0,0 +1,82 @@
+* Queue deadlocking: users have a max of 10 slots in the queue, with the current implementation one is taken up just to manage the jobs. Could lead to deadlock scenarios
+* Missing the ability to be able to run, manage and track jobs and their subjobs 
+* No good way to test and hard to benchmark or measure performance 
+* Code is split more than is necessary
+
+## Author(s)
+
+@leia-sefkin
+
+## Status
+
+Pending
+
+## Alternatives Considered
+
+* Writing a new Batch App
+* Implementing a Dynamic Service
+* Implementing a standalone Narrative Widget
+* Implementing a new (set) of EE2 endpoint(s)
+
+## Decision Outcome
+
+After much [discussion](https://docs.google.com/document/d/1PoiOas-hqgHONNzmCVupjmNCyvUvcned5YfZf-5CxsI/edit#) [and research](https://docs.google.com/spreadsheets/d/1FGecELaEBAWQ7ljXsR29h5RdtmYV_jMyLSiJAWcm4mE/edit#gid=492998772) the decision was to create a new endpoint (or set of endpoints) in EE2 for the purpose of running batch analysis workflows at KBase. 
+
+## Consequences
+
+We will have to implement a new narrative UI, however this was work that would happen regardless due as we are looking to improve the UX for batch upload and analysis at KBase. 
+
+As far as we have discussed, this decision only seems to have positive consequences. We will be able to streamline the code structure at KBase, make batch analysis easier to test and improve usability and job management. 
+
+Initial architecture sketches for what this could look like have been started [here](https://miro.com/app/board/o9J_kmb4y4Q=/?moveToWidget=3074457350208108487&cot=12)
+
+Still to be determined (not in scope of this ADR): 
+* What other endpoints we may need such as cancel batch
+* Other implementation details such as will the Narrative poll EE2 for updates on jobs or subscribe to Kafka. 
+
+## Pros and Cons of the Alternatives
+
+### Writing a new Batch App
+
+* `+` We have a potential first pass at implementation [here](https://github.com/bio-boris/simplebatch)
+* `+` Removes dependency on KB Parallel
+* `+` No longer uses up a slot in the queue
+* `+` Offers some improvement upon job management
+* `-` Code is still split between locations
+* `-` No good way to test and hard to benchmark or measure performance 
+* `-` Tracking and managing sub jobs is still a challenge
+
+### Implementing a Dynamic Service
+* `+` No longer uses an app
+* `+` Custom endpoint
+* `+` No longer uses a slot in the queue
+* `+` Can manage jobs more closely: have tables, more control over lifecycle, resource requests, canceling subjobs
+* `-` Hard to test (secret variables, dynamic services in app-dev and prod have same catalog, so hard to tell which service is where; service wizard can cause problems) 
+* `-` Takes microservices approach to the extreme - if we have a service for this it makes more sense for it to live within EE2 rather than be further separated
+* `-` Locked into KB-SDK
+* `-` Coupled with other dynamic services
```

Comment:
I don't quite understand this con - what does this mean?

Reply:
The dynamic services are also coupled with Rancher1, and DevOps can't easily modify the deployment of these without changes to the service wizard

Reply:
added this detail